File: cses/coinpiles.cpp
#include <algorithm>
#include <iostream>

void solve(int a, int b) {
    if (std::max(a, b) <= std::min(a, b) * 2) {
        if ((a + b) % 3 == 0) {
            std::cout << "YES" << '\n';
        } else {
            std::cout << "NO" << '\n';
        }
    } else {
        std::cout << "NO" << '\n';
    }
}

int main() {
    int t;
    std::cin >> t;
    while (t--) {
        int a, b;
        std::cin >> a >> b;
        solve(a, b);
    }
}

File: cses/coin-piles.py
def solve():
    a, b = map(int, input().split())

    if (a + b) % 3 != 0 or a < b // 2 or b < a // 2:
        print("NO")
    else:
        print("YES")


def main():
    t = int(input())
    for _ in range(t):
        solve()


if __name__ == "__main__":
    main()
